### ISSUE TITLE
Where is configuration stored?

### ISSUE THREAD

**Issue by simendsjo** [tridactyl/tridactyl#4094]
I managed to `:blacklistadd .`, and now everything is blacklisted, and Tridactyl is in effect disabled.
I'm unable to find out where Tridactyl stores these settings.
This may be more of a "Firefox extension" question, but I've googled and searched through the .mozilla folder without any luck.

**Comment by benoit-pierre** [comment]
You can see the list of auto-commands in the `userconfig.autocmds.DockStart` key of the extension local storage (accessible through `about:debug`). But there does not seem to be a way to edit said value (except for clearing the whole `userconfig` section).

You should be able to clear the blacklist entry by adding a `blacklistremove .` entry in your startup rc (in `~/.config/tridactyl/tridactylrc`) and restarting Firefox.

**Comment by simendsjo** [comment]
Adding `blacklistremove .` to the config file didn't seem to do anything. Tried finding the value in the filesystem (in a sqlite database..?), but I was unable to find it.

**Comment by benoit-pierre** [comment]
You need to make sure you're passing the same argument as when you used `blacklistadd`, so check the extension storage for the value you need to use.

**Comment by benoit-pierre** [comment]
Unfortunately, there does not seem to be a way to remove all auto-commands for a particular event.

**Comment by benoit-pierre** [comment]
It's possible to remove all the `DocStart` auto-commands by running this code in the extension debugging console:
```js
var storage = await browser.storage.local.get();
storage.userconfig.autocmds.DocStart = {};
await browser.storage.local.set(storage);
```

**Comment by simendsjo** [comment]
> It's possible to remove all the `DocStart` auto-commands by running this code in the extension debugging console:
> 
> ```js
> var storage = await browser.storage.local.get();
> storage.userconfig.autocmds.DocStart = {};
> await browser.storage.local.set(storage);
> ```

Thanks, works great! I would never have found this by myself, so a big thanks. I'll try not to completely brick the extension again :) Really great extension too!

**Comment by kohane27** [comment]
I was spending hours trying to debug why tridactyl is not reading my config. They are located in `~/.config/tridactyl/tridactylrc`. I even tried `~/.tridactylrc`. Then I realized I did not yet install [native_messenger](https://github.com/tridactyl/native_messenger), which includes doing the following:

> This small application allows Tridactyl to communicate with your system, allowing Tridactyl to:
>
>     read files on your computer, including "RC" configuration files

After successfully installing native native_messenger, my config is successfully loaded. Hopefully it helps.
